Packs/TrendMicroEmailSecurity/Integrations/TrendMicroEmailSecurityEventCollector/TrendMicroEmailSecurityEventCollector_test.py

```diff
@@ -0,0 +1,356 @@
+        ),
+    ],
+)
+def test_managing_set_last_run(
+    mocker,
+    mock_client: Client,
+    args,
+    mock_exception: NoContentException,
+    mock_api: dict,
+):
+    time_to = datetime.now()
+    mocker.patch.object(
+        mock_client,
+        "get_logs_request",
+        side_effect=mock_exception,
+        return_value=mock_api,
+    )
+    results = managing_set_last_run(
+        client=mock_client,
+        len_events=args["len_events"],
+        limit=args["limit"],
+        last_run=args["last_run"],
+        time_from=args["time_from"],
+        time_to=time_to,
+        event_type=args["event_type"],
+        next_token=args["next_token"],
+    )
+
+    expected_time_from_for_event_type = mock_api.get("logs", [{}])[0].get(
+        "genTime"
+    ) or (time_to + timedelta(seconds=1)).strftime(DATE_FORMAT_EVENT)
+    assert (
+        results.get(f"time_{args['event_type']}_from")
+        == expected_time_from_for_event_type
+    )
+
+
+@pytest.mark.parametrize(
+    "event_mock, next_token, limit",
+    [(({},), "abc%20abc", 1), (({"nextToken": "abc%20abc", "logs": [{}]}, {}), "", 2)],
+)
+def test_fetch_by_event_type_token_unquote(
+    mocker, mock_client: Client, event_mock: tuple[dict], next_token: str, limit: int
+):
+    """
+    Given:
+        - next_token with quote character
+    When:
+        - run fetch_by_event_type function
+    Then:
+        - Ensure that the next_token argument sent to the API request is unquoted
+    """
+    mock_api = mocker.patch.object(
+        mock_client, "get_logs_request", side_effect=event_mock
+    )
+    fetch_by_event_type(mock_client, "", "", limit, next_token, "", False)
+    assert mock_api.call_args[0][1]["token"] == "abc abc"
+
+
+@pytest.mark.parametrize(
+    "event_mock, limit",
+    [
+        (({},), 1),
+        (({"nextToken": "test", "logs": [{}]}, {}), 2),
+        (({"logs": [{}]},), 1),
+        (({"nextToken": "test", "logs": [{}]}, {"logs": [{}]}), 2),
+        ((NoContentException(),), 1),
+        (({"nextToken": "test", "logs": [{}]}, NoContentException()), 2),
+    ],
+)
+def test_fetch_by_event_type_returned_next_token_none(
+    mocker, mock_client: Client, event_mock: tuple[dict], limit: int
+):
+    """
+    Given:
+        - args for fetch_by_event_type function
+    When:
+        - run fetch_by_event_type function and return from api request response by event mock
+    Then:
+        - Ensure that if a response is returned that does not have the "logs" key
+          or the "nextToken" key or returned a NoContentException even if ift occurs
+          in the second iteration the function returns `next_token == None`
+    """
+    mocker.patch.object(mock_client, "get_logs_request", side_effect=event_mock)
+    _, next_token = fetch_by_event_type(mock_client, "", "", limit, None, "", False)
+
+    assert not next_token
+
+
+@pytest.mark.parametrize(
+    "event_mock, limit, expected_results",
+    [
```

Comment:
consider `pytest.param` here too, it's a bit difficult telling the different cases apart.